### ISSUE TITLE
$invalid is preserved when blur event occurs

### ISSUE THREAD

**Issue by OlexandrPopov** [angular-ui/ui-mask#118]
I use a phone mask for my input field. When I type a few digits of a phone number into the input and move the focus to another input, the masked input is cleared but it has invalid state.

Steps to reproduce:
1. Open this [plnkr](https://plnkr.co/edit/9uNby52iwAF0DOdhUWQZ?p=preview) 
2. Focus the input and type only a few digits of a phone number.
3. Click outside the input (make the input unfocused)
4. The input is empty but $invalid is true.

I use the latest version available in bower, probably it's 1.7.2.

**Comment by lukepfeiffer10** [comment]
Thanks for the plunkr and clear steps to follow. I'll look into it when I get a chance (hopefully soon).

**Comment by RafaelTaranto** [comment]
In this bit of code on the blurHandler function, using $setViewValue solves the bug. 
```
scope.$apply(function() {
    //don't call $setViewValue to avoid changing $pristine state.
    controller.$viewValue = '';
});
```
I'll not submit a PR right now because there is a comment explicitly saying not use $setViewValue. I'm not really sure why the component should keep it's $pristine state after a blur event.

Anyone knows the reason for this?

**Comment by lukepfeiffer10** [comment]
@RafaelTaranto Thanks for looking into this and I was about to link that issue for explanation until I saw your edit.

**Comment by lukepfeiffer10** [comment]
Any updates @RafaelTaranto?

**Comment by RafaelTaranto** [comment]
@lukepfeiffer10 Sorry, I got completly swamped at work..

If anyone wants to test and PR i'm using this code in the blurHandler function.

`scope.$apply(function() {
    if (!controller.$pristine) {
      controller.$setViewValue('');
    }
});`

Haven't noticed any side effects to this fix so far, but I'm not with a lot of time to test this for real.

**Comment by lukepfeiffer10** [comment]
Fixed with #134
